coreneuron/utils/reports/nrnsection_mapping.h

```diff
@@ -34,15 +35,15 @@ struct SecMapping {
     }
 
     /** @brief return total number of sections in section list */
-    size_t num_sections() {
+    size_t num_sections() const {
         return secmap.size();
     }
 
     /** @brief return number of segments in section list */
-    size_t num_segments() {
+    size_t num_segments() const {
         size_t count = 0;
-        for (secseg_it_type iterator = secmap.begin(); iterator != secmap.end(); iterator++) {
-            count += iterator->second.size();
+        for (auto & iterator : secmap) {
```

Comment:
```suggestion
        for (auto const& iterator : secmap) {
```

Reply:
IMO you can also use `std::accumulate` here?
Besides, you can use `auto` in argument types of lambda functions  in C++14 so that  something horrific like:

```cpp
[](size_t accu, std::map<..., ...>::const_iterator& item) { return accu + item->second.size(); }
```

becomes

```cpp
[](size_t accu, const auto& item) { return accu + item->second.size(); }
```

Reply:
Thank you for pointing this out, @tristan0x ! I somehow forgot that `auto` parameter types for lambdas were introduced in C++14. It looked too cumbersome without so I skipped it. Now it's pretty!